### DISCUSSION TITLE
Source, dest port of SD messages

### DISCUSSION THREAD

**Discussion by liveeasily** [COVESA/vsomeip#708] [Q&A] [answered]
Hello community, I am a beginner in vsomeip.

In udp_server_endpoint_impl.cpp, the unicast socket is always bound to the configuration's SD port. Therefore, the source port of the SD messages (e.g., FIND, OFFER, etc.) sent via multicast is the same as the SD port (=destination port).
Additionally, in routing_manager_impl.cpp, if the source port of a received SD message is different from the configuration's SD port, it is ignored.As a result, vsomeip assumes that the source and destination ports of SD multicast messages are always the same.

Is my understanding above correct?
If so, such a restriction may often conflict with switches that drop UDP packets with the same source and destination port as a defense against DDoS attacks. Is there such a requirement in the SOME/IP SD specification?

**Comment by srl100** [comment]
Have a look at https://www.autosar.org/fileadmin/standards/R22-11/FO/AUTOSAR_PRS_SOMEIPServiceDiscoveryProtocol.pdf

In that document PRS_SOMEIPSD_00600 and PRS_SOMEIPSD_00601 say that UDP datagrams _shall_ use SD_PORT and then goes on to describe SD_PORT as the "_UDP Port for SD Messages (30490 as default)_".

NB: PRS_SOMEIPSD_00602 appears to modify this "_shall_" behaviour of PRS_SOMEIPSD_00600 in the presence of "_SD Endpoint Option_".

**Reply by srl100** [reply, marked as answer]
@liveeasily Did that answer your question? If so please mark this as "Answered".

**Reply by liveeasily** [reply]
@srl100 
Thanks for the answer! Yes, "PRS_SOMEIPSD_00601" describes the source port of SD messages.

_[PRS_SOMEIPSD_00601] SD_PORT shall be used as the source port for SD Unicast/Multicast Messages.(RS_SOMEIPSD_00002, RS_SOMEIPSD_00003)_
